frontend/openchat-agent/src/services/dexes/index.ts

```diff
@@ -23,6 +23,23 @@ export class DexesAgent {
         );
     }
 
+    async canSwap(tokens: Set<string>): Promise<Set<string>> {
+        const allPools = await this._icpSwapIndexClient.getPools();
+
+        const available = new Set<string>();
+
+        for (const p of allPools) {
+            if (tokens.has(p.token0)) {
```

Comment:
I think this should be `if (tokens.has(p.token0) && tokens.has(p.token1)) { // add both }`

Reply:
Since otherwise a token may only be swappable with another token which we don't support.